spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```

Comment:
- [ ] Also, another thing I noticed is that you have an unexpected nesting in both your tests for `Posts` and `Users`:

![image](https://github.com/rugwizangoga/Blog-app/assets/77069011/c2cc0316-0c20-4ee1-8cf6-c74e09c7a511)

Which, in code, looks like this:

```rb
describe 'GET #index' do
  ...
  describe 'GET #show' do
    ...
  end
end
```

Please consider fixing this structure to something like this:

```rb
describe 'GET #index' do
  ...
end

describe 'GET #show' do
  ...
end
```

Remember that both blocks are testing different requests, so it makes sense not to nest them! 🤓 🔧